### DISCUSSION TITLE
How to learn Mojo

### DISCUSSION THREAD

**Discussion by s3drik** [modular/modular#394] [General]
I'm python programmer, i know something about it i write some programs i love it. But i saw mojo and i wanted to code in mojo. there is only docs but in docs there things that i've never used: alias, pointers, heap, ... I think things like this used in C should i learn C or other compiled language for understanding mojo? And also english isn't my native language is there any easier docs

**Comment by cjohnson318** [comment]
If you sign up via email and get into the playground, then there are several ipython notebook files with example code that you can try out. My approach, as a Python developer, is to write something "the Python way" and then experiment with adding Mojo features or constructs and timing the results. Good luck!

**Comment by mojodojodev** [comment]
Hi @Saddibek1 yes there aren't any introductory materials for Python programmers yet, I'm trying to remedy this but only at the very early stages: [Intro to Mojo](https://mojodojo.dev/guides/intro-to-mojo/setup.html). Would love your feedback if you get a chance, it's only a tiny amount of everything you'll need to learn at this stage though.

Learning another language would be a good strategy yes and C is a great choice, but perhaps even better would be Swift or Rust because they have good modern solutions for memory safety though `ownership` and `lifetimes`, Mojo will make a lot more sense after learning one of those. I haven't used Swift so can't comment on the introductory materials, but Rust has a great book for newcomers called [The Rust Book](https://doc.rust-lang.org/book/ch01-00-getting-started.html), which you can supplement with [Rustlings](https://github.com/rust-lang/rustlings) for fun exercises.

**Reply by s3drik** [reply]
Well i started to learn rust. But there is another question. Is mojo faster than rust?

**Reply by mojodojodev** [reply]
It adds a couple things that will make it faster in some scenarios, e.g. autotuning to optimize for different hardware and allowing to declare `@register_passable` for small structs might have a significant impact. All of these compiled languages like C/C++/Rust/Mojo have almost no overhead and for CPU are optimized through LLVM, so should only be limited by how you write your code.
